test/test_primitives.cpp

```diff
@@ -235,3 +235,12 @@ TORRENT_TEST(client_data_initialize)
 	TEST_CHECK(v.get<int>() == &a);
 	TEST_CHECK(*v.get<int>() == 1337);
 }
+
+TORRENT_TEST(announce_endpoint_initialize)
+{
+	// announce_endpoint has an array of announce_infohash and
+	// announce_infohash has the constructor is marked TORRENT_UNEXPORT
+	// it's important that announce_endpoint provides a constructor
+	announce_endpoint ae;
+	TEST_EQUAL(ae.enabled, false);
```

Comment:
this defaults to `true`

Reply:
it defaults to `false` in my local env (macOS with clang) and for what I see, in a few other travis jobs. Do you want to make it `true` explicitly?

Reply:
oh, I was looking [here](https://github.com/arvidn/libtorrent/blob/RC_1_2/src/announce_entry.cpp#L58). but it seems the other constructors don't initialize it. I think it should be initialized in-class. where the member is declared.

Reply:
wait.. why is this default constructible? does it have to be because it's in a container?
If it doesn't have to be, I think it would be best to delete the default constructor. if it has to be, then a default constructor would have to be defined that initializes all these members. in-class initializers don't work on bitfields

Reply:
I'm also a bit confused because I was looking at `RC_1_2` instead of `RC_2_0` :)

Reply:
you know what. in `RC_2_0` I finally separated the public announce entry types from the internal ones. So there really isn't any good reason for `disabled` to be a single bit in the public struct. It could just be a `bool` and you could get a proper in-class initializer for it.